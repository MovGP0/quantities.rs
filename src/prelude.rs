// ---------------------------------------------------------------------------
// Copyright:   (c) 2021 ff. Michael Amrhein (michael@adrhinum.de)
// License:     This program is part of a larger application. For license
//              details please read the file LICENSE.TXT provided together
//              with the application.
// ---------------------------------------------------------------------------
// $Source$
// $Revision$

//! This module reexports all macros and types needed to define a quantity.

pub use crate::{Amnt, AmountT, Quantity, SIPrefix, Unit, Unitless};
#[cfg(feature = "fpdec")]
pub use crate::{Dec, Decimal};
pub use core::cmp::Ordering;
pub use core::fmt;
pub use core::ops::{Add, Div, Mul, Sub};
pub use qty_macros::quantity;
