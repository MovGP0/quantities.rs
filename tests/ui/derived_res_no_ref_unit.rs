// ---------------------------------------------------------------------------
// Copyright:   (c) 2022 ff. Michael Amrhein (michael@adrhinum.de)
// License:     This program is part of a larger application. For license
//              details please read the file LICENSE.TXT provided together
//              with the application.
// ---------------------------------------------------------------------------
// $Source$
// $Revision$

use quantities::prelude::*;

#[quantity]
#[ref_unit(Flop, "f")]
#[unit(Kiloflop, "kf", 1000., "1000路f")]
struct Foo {}

#[quantity]
#[ref_unit(Emil, "e")]
#[unit(Milliemil, "me", 0.001, "0.001路e")]
struct Bar {}

#[quantity(Foo * Bar)]
#[unit(Bazoo, "b", "1路f路e")]
struct Baz {}

fn main() {}
