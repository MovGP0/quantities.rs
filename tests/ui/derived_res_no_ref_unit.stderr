error[E0277]: the trait bound `Baz: quantities::HasRefUnit` is not satisfied
  --> tests/ui/derived_res_no_ref_unit.rs:22:1
   |
22 | #[quantity(Foo * Bar)]
   | ^^^^^^^^^^^^^^^^^^^^^^ the trait `quantities::HasRefUnit` is not implemented for `Baz`
   |
   = help: see issue #48214
   = note: this error originates in the attribute macro `quantity` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Baz: quantities::HasRefUnit` is not satisfied
  --> tests/ui/derived_res_no_ref_unit.rs:22:1
   |
22 | #[quantity(Foo * Bar)]
   | ^^^^^^^^^^^^^^^^^^^^^^ the trait `quantities::HasRefUnit` is not implemented for `Baz`
   |
   = note: required because of the requirements on the impl of `std::ops::Div<Bar>` for `Baz`
   = help: see issue #48214
   = note: this error originates in the attribute macro `quantity` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Baz: quantities::HasRefUnit` is not satisfied
  --> tests/ui/derived_res_no_ref_unit.rs:22:1
   |
22 | #[quantity(Foo * Bar)]
   | ^^^^^^^^^^^^^^^^^^^^^^ the trait `quantities::HasRefUnit` is not implemented for `Baz`
   |
   = note: required because of the requirements on the impl of `std::ops::Div<Foo>` for `Baz`
   = help: see issue #48214
   = note: this error originates in the attribute macro `quantity` (in Nightly builds, run with -Z macro-backtrace for more info)
